time_tracker_api/activities/activities_namespace.py

```diff
@@ -41,41 +43,49 @@
     activity_response_fields
 )
 
-activities_dao = create_dao()
+activity_dao = create_dao()
 
 
 @ns.route('')
 class Activities(Resource):
     @ns.doc('list_activities')
-    @ns.marshal_list_with(activity, code=200)
+    @ns.marshal_list_with(activity, code=HTTPStatus.OK)
     def get(self):
-        return []
+        """List all activities"""
+        return activity_dao.get_all(), HTTPStatus.OK
 
     @ns.doc('create_activity')
+    @ns.response(HTTPStatus.BAD_REQUEST, 'Bad request')
     @ns.expect(activity_input)
-    @ns.marshal_with(activity, code=201)
-    @ns.response(400, 'Invalid format of the attributes of the activity.')
+    @ns.marshal_with(activity, code=HTTPStatus.CREATED)
     def post(self):
-        return ns.payload, 201
+        """Create an activity"""
+        return activity_dao.create(ns.payload), HTTPStatus.CREATED
 
 
 @ns.route('/<string:id>')
-@ns.response(404, 'Activity not found')
-@ns.param('id', 'The unique identifier of the activity')
+@ns.response(HTTPStatus.NOT_FOUND, 'Activity not found')
+@ns.param('id', 'The activity identifier')
 class Activity(Resource):
     @ns.doc('get_activity')
-    @ns.marshal_with(activity)
+    @ns.response(HTTPStatus.UNPROCESSABLE_ENTITY, 'The id has an invalid format')
+    @ns.marshal_with(activity, code=HTTPStatus.OK)
     def get(self, id):
-        return {}
-
-    @ns.doc('delete_activity')
-    @ns.response(204, 'The activity was deleted successfully (No content is returned)')
-    def delete(self, id):
-        return None, 204
+        """Get an activity"""
+        return activity_dao.get(id), HTTPStatus.OK
 
-    @ns.doc('put_activity')
-    @ns.response(400, 'Invalid format of the attributes of the activity.')
+    @ns.doc('update_activity')
+    @ns.response(HTTPStatus.UNPROCESSABLE_ENTITY, 'The data has an invalid format.')
```

Comment:
I would suggest to remove `UNPROCESSABLE_ENTITY` (it would be in the class) and add:
```
    @ns.response(HTTPStatus.BAD_REQUEST, 'Invalid format or structure of the activity')
    @ns.response(HTTPStatus.CONFLICT, 'An activity already exists with this new data')
```